python/xorbits/_mars/dataframe/initializer.py

```diff
@@ -109,6 +122,8 @@ def __init__(
                 if copy:
                     pdf = pdf.copy()
             else:
+                if isinstance(data, list):
```

Comment:
This branch is the "default" branch, which should handle all but the special cases above.

Execute and fetch data recursively at the very begining does make sense, but is it possible that the `data` is a dict? 

For example,
```python
data = {"max_col1": df["col1"].max(), "max_col2": df["col2"].max()}
```

And what if the elements are not TensorData (could be series)?
For example,
```python
data = [[df["col1"], df["col2"]]]
```

Reply:
I am converting list of list to dict of list type since we can handle these cases you mention here when data is dict type